### PR TITLE
installer workflow: switch tls generation from terraform to go 

### PR DESCRIPTION
**Pull request by yifan-gu**
This PR switch the tls generate from using terraform module to using go code.
This will further remove our dependency on terraform (which our goal is to remove any assets generation job off terraform).

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
@yifan-gu should we also remove all the `terraform code` regarding `tls` in this pr?

**Comment by yifan-gu** [comment]
@abhinavdahiya ok will do.

File: installer/pkg/config-generator/tls.go [hunk 1]
```diff
@@ -91,6 +91,15 @@ func (c *ConfigGenerator) GenerateTLSConfig(clusterDir string) error {
 		return fmt.Errorf("failed to generate etcd CA: %v", err)
 	}
 
+	err = copyFile(filepath.Join(clusterDir, etcdCAKeyPath), filepath.Join(clusterDir, "generated/tls/etcd-client-ca.key"))
+	if err != nil {
+		return fmt.Errorf("failed to import kube CA cert into ingress-ca.crt: %v", err)
```
Comment (abhinavdahiya) [inline installer/pkg/config-generator/tls.go]: error doesn't match

File: installer/pkg/config-generator/tls.go [hunk 2]
```diff
@@ -91,6 +91,15 @@ func (c *ConfigGenerator) GenerateTLSConfig(clusterDir string) error {
...
+	}
+	err = copyFile(filepath.Join(clusterDir, etcdCACertPath), filepath.Join(clusterDir, "generated/tls/etcd-client-ca.crt"))
+	if err != nil {
+		return fmt.Errorf("failed to import kube CA cert into ingress-ca.crt: %v", err)
```
Comment (abhinavdahiya) [inline installer/pkg/config-generator/tls.go]: error doens't match

File: installer/pkg/config-generator/tls.go [hunk 3]
```diff
@@ -85,46 +86,55 @@ func (c *ConfigGenerator) GenerateTLSConfig(clusterDir string) error {
...
 	// generate aggregator CA
 	cfg = &tls.CertCfg{
 		Subject:   pkix.Name{CommonName: "aggregator", OrganizationalUnit: []string{"bootkube"}},
 		KeyUsages: x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
-		Validity:  validityThreeYears,
+		Validity:  validityTenYears,
 		IsCA:      true,
 	}
-	if _, _, err := generateCert(clusterDir, caKey, caCert, aggregatorCAKeyPath, aggregatorCACertPath, cfg); err != nil {
+	if _, _, err := generateCert(clusterDir, caKey, caCert, aggregatorCAKeyPath, aggregatorCACertPath, cfg, false); err != nil {
```
Comment (abhinavdahiya) [inline installer/pkg/config-generator/tls.go]: aggregator ca is used to sign certs. shouldn't ignore.

File: installer/pkg/config-generator/tls.go [hunk 4]
```diff
@@ -208,11 +218,11 @@ func (c *ConfigGenerator) GenerateTLSConfig(clusterDir string) error {
 		KeyUsages:    x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
 		ExtKeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
 		Subject:      pkix.Name{CommonName: "kube-apiserver-proxy", Organization: []string{"kube-master"}},
-		Validity:     validityThreeYears,
+		Validity:     validityTenYears,
 		IsCA:         false,
 	}
 
-	if _, _, err := generateCert(clusterDir, kubeCAKey, kubeCACert, apiServerProxyKeyPath, apiServerProxyCertPath, cfg); err != nil {
+	if _, _, err := generateCert(clusterDir, kubeCAKey, kubeCACert, apiServerProxyKeyPath, apiServerProxyCertPath, cfg, false); err != nil {
```
Comment (abhinavdahiya) [inline installer/pkg/config-generator/tls.go]: should be signed by `aggregator ca`

File: installer/pkg/config-generator/tls.go [hunk 5]
```diff
@@ -194,12 +204,12 @@ func (c *ConfigGenerator) GenerateTLSConfig(clusterDir string) error {
...
 			"openshift-apiserver.kube-system.svc.cluster.local",
 			"localhost", "127.0.0.1"},
-		Validity:    validityThreeYears,
+		Validity:    validityTenYears,
 		IPAddresses: []net.IP{net.ParseIP(apiServerAddress)},
 		IsCA:        false,
 	}
 
-	if _, _, err := generateCert(clusterDir, kubeCAKey, kubeCACert, osAPIServerKeyPath, osAPIServerCertPath, cfg); err != nil {
+	if _, _, err := generateCert(clusterDir, kubeCAKey, kubeCACert, osAPIServerKeyPath, osAPIServerCertPath, cfg, true); err != nil {
```
Comment (abhinavdahiya) [inline installer/pkg/config-generator/tls.go]: should be signed by `aggregatgor ca`

File: installer/pkg/config-generator/tls.go [hunk 6]
```diff
@@ -137,11 +147,11 @@ func (c *ConfigGenerator) GenerateTLSConfig(clusterDir string) error {
 			fmt.Sprintf("%s.%s", "*", baseAddress),
 		},
 		Subject:  pkix.Name{CommonName: baseAddress, Organization: []string{"ingress"}},
-		Validity: validityThreeYears,
+		Validity: validityTenYears,
 		IsCA:     false,
 	}
 
-	if _, _, err := generateCert(clusterDir, kubeCAKey, kubeCACert, ingressKeyPath, ingressCertPath, cfg); err != nil {
+	if _, _, err := generateCert(clusterDir, kubeCAKey, kubeCACert, ingressKeyPath, ingressCertPath, cfg, false); err != nil {
```
Comment (abhinavdahiya) [inline installer/pkg/config-generator/tls.go]: might be useful to set true.

**Pull request merged**